pkg/controller/vsphere/reconciler.go

```diff
@@ -694,6 +695,7 @@ func clone(s *machineScope) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("error triggering clone op for machine %v: %w", s, err)
 	}
+	task.Wait(s.Context)
```

Comment:
Hm, will this block until machine clone to be completed?

Reply:
Yes. I think need to additionally figure out how it might impact reconciliation. I looked around a bit, don't think that it will break machine creation, but actual creation flow will changed a bit anyway.
Want to mention that we have a bunch of logic around checking task status - https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/reconciler.go#L97

Reply:
good catch! yeah I'm currently looking into getting rid of this. looks like I was working around a lower-level issue in govmomi simulator.

Reply:
https://github.com/openshift/machine-api-operator/pull/995/commits/d40c664dd14ba51fe22cb0b1955634d188a658ae should address this concern. `TestCreate` will need to wait for the simulator task to complete since its a goroutine (as mentioned before, this is new behavior in govmomi 0.25+); otherwise, the unit test will proceed with teardown operations _while_ the clone is still completing which results in a panic.